static/src/javascripts/projects/commercial/modules/dfp/create-slot.js

```diff
@@ -0,0 +1,144 @@
+        name: 'merchandising-high',
+        sizeMappings: {
+            mobile: [
+                adSizes.outOfPage,
+                adSizes.empty,
+                adSizes.merchandisingHigh,
+                adSizes.fluid,
+            ],
+        },
+    },
+    'high-merch-lucky': {
+        label: false,
+        refresh: false,
+        name: 'merchandising-high-lucky',
+        sizeMappings: {
+            mobile: [adSizes.outOfPage, adSizes.empty, adSizes.fluid],
+        },
+    },
+    'high-merch-paid': {
+        label: false,
+        refresh: false,
+        name: 'merchandising-high',
+        sizeMappings: {
+            mobile: [
+                adSizes.outOfPage,
+                adSizes.empty,
+                adSizes.merchandisingHighAdFeature,
+                adSizes.fluid,
+            ],
+        },
+    },
+    inline: inlineDefinition,
+    mostpop: inlineDefinition,
+    comments: inlineDefinition,
+    'top-above-nav': {
+        sizeMappings: {
+            mobile: [
+                adSizes.outOfPage,
+                adSizes.empty,
+                adSizes.mpu,
+                adSizes.fabric,
+                adSizes.fluid,
+            ],
+        },
+    },
+};
+
+const createAdSlotElement = (
+    name: string,
+    attrs: Object,
+    classes: Array<string>
+) => {
+    const adSlot: HTMLDivElement = document.createElement('div');
+    adSlot.id = `dfp-ad--${name}`;
+    adSlot.className = `js-ad-slot ad-slot ${classes.join(' ')}`;
+    adSlot.setAttribute('data-link-name', `ad slot ${name}`);
+    adSlot.setAttribute('data-name', name);
+    Object.keys(attrs).forEach(attr => {
+        adSlot.setAttribute(attr, attrs[attr]);
+    });
+    return adSlot;
+};
+
+const createSlot = (type: string, options: Object = {}) => {
+    const attributes = {};
+    const definition: Object = adSlotDefinitions[type];
+    const slotName = options.name || definition.name || type;
+    const classes = options.classes
+        ? options.classes.split(' ').map(cn => `ad-slot--${cn}`)
+        : [];
+
+    const sizes = Object.assign({}, definition.sizeMappings);
+
+    if (options.sizes) {
+        Object.keys(options.sizes).forEach(size => {
+            if (sizes[size]) {
+                sizes[size] = sizes[size].concat(options.sizes[size]);
+            } else {
+                sizes[size] = options.sizes[size];
+            }
+        });
+    }
+
+    Object.keys(sizes).forEach(size => {
+        sizes[size] = sizes[size].join('|');
+    });
+
+    Object.assign(attributes, sizes);
+
+    if (definition.label === false) {
+        attributes.label = 'false';
+    }
+
+    if (definition.refresh === false) {
+        attributes.refresh = 'false';
+    }
+
+    classes.push(`ad-slot--${slotName}`);
+
+    return createAdSlotElement(
+        slotName,
+        Object.entries(attributes).reduce(
+            (result, [key, value]) =>
+                Object.assign({}, result, { [`data-${key}`]: value }),
```

Comment:
[Long story short](https://github.com/airbnb/javascript/issues/719): the way this was originally written fell foul of the linter, specifically `no-param-reassign`, since the accumulator `result` was being extended.

I've opted to create a new Object in each iteration, it is more functional and less efficient...but we are dealing with an order of ~10 max attributes i.e. who cares.